appservice/workers/transaction_scheduler.go

```diff
@@ -203,7 +203,7 @@ func send(
 	transaction []byte,
 ) error {
 	// POST a transaction to our AS
-	address := fmt.Sprintf("%s/transactions/%d", appservice.URL, txnID)
+	address := fmt.Sprintf("%s/transactions/%d?access_token=%s", appservice.URL, txnID, appservice.HSToken)
 	resp, err := client.Post(address, "application/json", bytes.NewBuffer(transaction))
```

Comment:
This should be PUT according to https://matrix.org/docs/spec/application_service/r0.1.2#put-matrix-app-v1-transactions-txnid